main.c

```diff
@@ -677,6 +677,24 @@ static bool should_stay_in_bootloader()
 	return  wd_says_so;
 }
 
+static void jump_to_app(struct image_header *hdr)
+{
+	if (!should_stay_in_bootloader() && image_header_ok(hdr)) {
+
+		uint32_t vtor = *((uint32_t *)(XIP_BASE + IMAGE_HEADER_OFFSET));
+		disable_interrupts();
+		reset_peripherals();
+		jump_to_vtor(vtor);
+	}
+}
+
+int64_t alarm_callback(alarm_id_t id, void *user_data) {
+    printf("Timer %d fired!\n", (int) id);
```

Comment:
Please check if we can use this one instead inside this callback `DBG_PRINTF`

Reply:
It can be used, DBG_PRINTF enables debug data to be sent via USB of the Pico not UART. 

USB disconnects and reconnects when mcu is reset and need to have a delay so that the terminal can process it. Refer jira ticket for the image of the debug line.

Alternatively we can send it to the UART line via uart_puts() API and view the same.

Reply:
To use the DBG_PRINTF we also need to `#define DEBUG` in the beginning

Reply:
DEBUG should not be used for production eventually, thus it should not be hard coded. however it need to be passed  as a compile option while building the project so that we can use debug functionality while developing

Reply:
Kindly approach me to see how we will move forward